jmxfetch.py

```diff
@@ -238,153 +306,44 @@ def is_jmx_check(cls, check_config, check_name, checks_list):
+    agentConfig = get_config(parse_args=False, cfg_path=config_path)
+    osname = get_os()
+    try:
+        confd_path = get_confd_path(osname)
+    except PathNotFound, e:
+        log.error("No conf.d folder found at '%s' or in the directory where"
+                  "the Agent is currently deployed.\n" % e.args[0])
 
-    @classmethod
-    def get_path_to_jmxfetch(cls):
-        if get_os() != 'windows':
-            return os.path.realpath(os.path.join(os.path.abspath(__file__), "..", "checks", "libs", JMX_FETCH_JAR_NAME))
+    return confd_path, agentConfig
 
-        return os.path.realpath(os.path.join(os.path.abspath(__file__), "..", "..", "jmxfetch", JMX_FETCH_JAR_NAME))
 
-    @classmethod
-    def start(cls, confd_path, agentConfig, logging_config, path_to_java, java_run_opts,
-        default_check_frequency, jmx_checks, command, reporter, tools_jar_path):
-        statsd_port = agentConfig.get('dogstatsd_port', "8125")
+def main(config_path=None):
+    """ JMXFetch main entry point """
+    confd_path, agentConfig = init(config_path)
 
-        if reporter is None:
-            reporter = "statsd:%s" % str(statsd_port)
-
-        log.info("Starting jmxfetch:")
-        jmx_connector_pid = None
-        try:
-            path_to_java = path_to_java or "java"
-            java_run_opts = java_run_opts or ""
-            path_to_jmxfetch = JMXFetch.get_path_to_jmxfetch()
-            path_to_status_file = os.path.join(tempfile.gettempdir(), "jmx_status.yaml")
+    jmx_daemon = JMXFetch(confd_path, agentConfig)
+    # Run JMXFetch
+    jmx_daemon.run()
 
-            if tools_jar_path is None:
-                classpath = path_to_jmxfetch
-            else:
-                classpath = r"%s:%s" % (tools_jar_path, path_to_jmxfetch)
-
-            subprocess_args = [
-                path_to_java, # Path to the java bin
-                '-classpath',
-                classpath,
-                JMXFETCH_MAIN_CLASS,
-                '--check_period', str(default_check_frequency * 1000),  # Period of the main loop of jmxfetch in ms
-                '--conf_directory', r"%s" % confd_path, # Path of the conf.d directory that will be read by jmxfetch,
-                '--log_level', JAVA_LOGGING_LEVEL.get(logging_config.get("log_level"), "INFO"),  # Log Level: Mapping from Python log level to log4j log levels
-                '--log_location', r"%s" % logging_config.get('jmxfetch_log_file'), # Path of the log file
-                '--reporter',  reporter, # Reporter to use
-                '--status_location', r"%s" % path_to_status_file, # Path to the status file to write
-                command, # Name of the command
-            ]
-
-            subprocess_args.insert(4, '--check')
-            for check in jmx_checks:
-                subprocess_args.insert(5, check)
-
-            # Specify a maximum memory allocation pool for the JVM
-            if "Xmx" not in java_run_opts and "XX:MaxHeapSize" not in java_run_opts:
```

Comment:
Is it expected that you are removing these lines ? i don't see them being readded.